packages/web-components/src/components/ic-text-field/ic-text-field.tsx

```diff
@@ -56,6 +56,9 @@ export class TextField {
   @Element() el: HTMLIcTextFieldElement;
 
   @State() numChars: number = 0;
+  @State() maxCharactersReached: boolean = false;
+  @State() maxCharactersError: boolean = false;
```

Comment:
I'm a bit confused why these two state values appear to be doing a similar thing. They also appear to be based off one another. Is it necessary for both of these to be considered `state`?

Reply:
I needed separate states to control the error state occurring and the maxlength being applied to the input, without these the error state is applied once the maxCharacters is reached, when we only want it to be applied once the user attempts to exceed the maxLength.